Library/AddOn.cs

```diff
@@ -176,13 +204,27 @@ internal override void ReadXml(XmlTextReader reader)
             }
         }
 
-        internal override void WriteXml(XmlTextWriter xmlWriter)
+        internal void WriteItemBackedAddOnXml(XmlTextWriter xmlWriter)
         {
             xmlWriter.WriteStartElement("add_on");
+            
```

Comment:
Given the following update request:
```
addon.UnitAmountInCents["USD"] = 400;
addon.Update(item.ItemCode);
```

request xml is always 

```
<add_on>
  <unit_amount_in_cents>
    <USD>400</USD>
  </unit_amount_in_cents>
  <default_quantity>1</default_quantity>
  <display_quantity_on_hosted_page>false</display_quantity_on_hosted_page>
  <optional>true</optional>
</add_on>
```
when it should be
```
<add_on>
  <unit_amount_in_cents>
    <USD>400</USD>
  </unit_amount_in_cents>
</add_on>
```